src/SelfInitializingFakes/SelfInitializingFake.of.T.cs

```diff
@@ -64,6 +69,20 @@ internal SelfInitializingFake(Func<TService> serviceFactory, IRecordedCallReposi
             }
         }
 
+        Action<IFakeOptions<TService>> CreateWrapper(TService wrappedService)
+        {
+            return options =>
+            {
+#if FEATURE_NETCORE_REFLECTION
+                var optionsType = options.GetType().GetTypeInfo();
+#else
+                var optionsType = options.GetType();
+#endif
+                var wrapping = optionsType.GetMethod("Wrapping", new[] { typeof(TService) });
+                wrapping.Invoke(options, new object[] { wrappedService });

```

Comment:
Um. I don't see `Delegate.CreateDelegate` available in .NET Standard 1.6. Is it? Am I chasing the wrong approach?